hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/InjectedSqlClientPool.java

```diff
@@ -0,0 +1,48 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.pool.impl;
+
+import org.hibernate.engine.jdbc.spi.SqlStatementLogger;
+
+import io.vertx.sqlclient.Pool;
+
+/**
+ * A pool of reactive connections backed by a Vert.x {@link Pool}.
+ * <p>
+ * This is an alternative to using {@see ManagedSqlClientPool}, for
+ * the case in which one doesn't want Hibernate ORM to manage
+ * the lifecycle of the underlying pool.
+ * <p>
+ * This implementation is meant to be used in Quarkus.
+ */
+public final class InjectedSqlClientPool extends BaseSqlClientPool {
```

Comment:
Does this class actually have anything to do with dependency injection? If not, then I would like to find a different name for it. `CustomSqlClientPool` or `ExternalSqlClientPool` or `ContainerSqlClientPool` or something like that.

Reply:
+1 for `ExternalSqlClientPool`. I'd avoid the other two as custom doesn't express much, and Quarkus isn't a "container" in the classical sense.